--- Makefile.in.orig	2011-05-01 20:30:09.000000000 +0900
+++ Makefile.in	2011-05-01 20:30:43.000000000 +0900
@@ -130,7 +130,7 @@
 
 NOSEARCH = style/.nosearch
 
-all: lisp docs
+all: lisp
 	$(DESCEND)
 
 .PHONY: all lisp info dvi some install install-el install-lisp \
@@ -189,7 +189,7 @@
 
 some:	$(AUCELC) $(STYLEELC)
 
-install: install-lisp install-docs install-images install-startup
+install: install-lisp install-images install-startup
 	$(DESCEND)
 	$(MAKE) 'DESTDIR=$(DESTDIR)' install-metadata
 
